### ISSUE TITLE
Sequence either gets filled completely (anticlockwise) or dies out (clockwise)

### ISSUE THREAD

**Issue by mrmitchley** [TomWhitwell/TuringMachine#211]
Built a TM and a vactrol mix, and it was functional (calibrated, correct outputs, responded to CV on all inputs), but had the following issues:

1. Clock LED was not working
2. Accidentally chose super long 2x6 male pins, so the two boards were in contact, but were not mated nicely - I also didn't have m3 nuts, but the module was working, so..

I built a Pulses, so took the opportunity to reflow the resistor closest to the LED. On restoring it to the case, I found the following strange behaviour

1. When the main knob is CCW, the sequence is always on, then off (eg 8 on, 8 off). Introducing any randomness or flipping the write switch in either direction introduces a change, but within 2 or 3 cycles it settles back to on, then off.
2. When the main knob is clockwise, the sequence is always off. Any randomness introduced or flipping of the write switch quickly dies out within three or four cycles, always dying out in a similar sequence (eg 8 on, then 5 on, then 3 on, then 2 on, then nothing, always clumped together)
3. The clock LED still doesn't work, but the clock input does advance it correctly
4. The boards are now somewhat secured with m3 nuts on the spacers, but now it doesn't work, so...

**Comment by mrmitchley** [comment]
3. I examined the clock LED with a loupe and spotted the trace lifted, so I put in a connecting wire to the resistor, the clock LED works now.

4. I clipped the too-long pins, and the boards are now neatly mated together and held securely with the spacers.

The shift register however is still defaulting to on, then off (CCW) or off (CW). On power-up, the outputs when fully clockwise are all 1s, but any perturbation to this will cause the sequence to die out.

I've tried it out with two different power cables, and two different cases

**Comment by mrmitchley** [comment]
I've also tried swapping all the paired chips, no change in behaviour. 

Noise output is good, CV and pulse output are consistent with the LED pattern, module is fully random when at 12, and sequence lengths are correct.

**Comment by TomWhitwell** [comment]
So the problem may be in the shift registers - 4015 or the analog switches that control the shift register - 4016 - although the CCW suggests they're OK. A bad connection would probably give these symptoms. The binary sequence passes from the back to the front (the rotary switch) and back again.

**Comment by mrmitchley** [comment]
I took it out and jiggled it all, and the behaviour is now slightly better (seems to be dropping a bit at position 7/8) so I think bad connection is likely the culprit here. I may need to replace the pin headers and reflow the chip sockets.

For the doubled sequence lengths of the counter-clockwise position, is there any difference between how the first half of the sequence and the second half are handled? The second half always seems to be zeroed off.

**Comment by mrmitchley** [comment]
OK, the issue with the CW behavior is definitely in part a loose connection between the two boards, I've solved it by making sure they are very tightly pressed together, and I swapped the shift registers

The issue with the CCW behavior is still there, so perhaps its one shift register thats a little buggy (bad connection? issue with the board connection exacerbated by the back and forth? unsure).

I'm going to go ahead and mark this as closed, with the warning for future travelers: make sure your male headers are all the right size.
